src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -80,25 +80,31 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
 
             using var result = await selectCommand.ExecuteReaderAsync();
 
-            // at this point the query succeeded and the table exists
-            return DbConnectionValidatorResult.DocumentFound;
+            // At this point, the query work and the 'Document' table exists.
+            if (result.HasRows)
+            {
+                // At this point we know that the Document table and the ShellDescriptor record exists.
+                return DbConnectionValidatorResult.ShellDescriptorDocumentFound;
+            }
+
+            // At this point the Document table exists with no ShellDescriptor document.
+            return DbConnectionValidatorResult.DocumentTableFound;
```

Comment:
This is wrong, when the document table exists but without ShellDescriptor (e.g. if using shell settings from database), the validation should pass, based on the current code here we would need to return DocumentTableNotFound as done line 96 in the catch, that's why when I named it .Success I was just using one line at the very end but not here and in the catch.

Or line 87 return DocumentTableFound so that you don't have to check ShellDescriptorDocumentFound in tenant validator and setup service, so remove this state and here return a new ShellDescriptorDocumentNotFound state that will let the validation pass.